src/core/src/bootstrap/EnvLayer.py

```diff
@@ -64,20 +64,25 @@ def get_package_manager(self):
         """ Detects package manager type """
         ret = None
 
-        # choose default - almost surely one will match.
-        for b in ('apt-get', 'yum', 'zypper'):
-            code, out = self.run_command_output('which ' + b, False, False)
+        if self.platform.linux_distribution()[0] == Constants.AZURE_LINUX:
+            code, out = self.run_command_output('which tdnf', False, False)
             if code == 0:
-                ret = b
-                if ret == 'apt-get':
-                    ret = Constants.APT
-                    break
-                if ret == 'yum':
-                    ret = Constants.YUM
-                    break
-                if ret == 'zypper':
-                    ret = Constants.ZYPPER
-                    break
+                ret = Constants.TDNF
+        else:
+            # choose default - almost surely one will match.
+            for b in ('apt-get', 'yum', 'zypper'):
+                code, out = self.run_command_output('which ' + b, False, False)
+                if code == 0:
+                    ret = b
+                    if ret == 'apt-get':
```

Comment:
Does python have a switch statement equivalent?

Reply:
Also, why can't we just return the constant here, instead of having the break and then the return at the end? is that a python-ism I don't understand?

Reply:
Python doesn't have a dedicated switch statement, it can be implemented using other approaches, say, if, else ifs. I believe, the break statements and return at the end is to avoid having multiple return statements throughout the function, making it easier for code readability. @kjohn-msft, the code author, for further inputs, if any. Also, Koshy, is there any specific reason we have used only ifs within here instead of if and else ifs?

Reply:
With the breaks, if / elifs all behave the same in units of compute expended (either choice is inconsequential for this path). For other paths, elifs are preferred.

Reply:
Apparently python does have a switch-type statement (they call it match and it functions identically to a switch). I was mostly asking as a curiosity because this chunk of code looks like a switch

Reply:
All our code needs to be Python 2.7 (even though deprecated) and various versions of Python 3 compatible. So we can only use constructs that are common across the board or we need to custom build constructs that allow language-version agnostic behaviors. An example of this is backporting enum support with custom code so we could benefit from cleaner code (see Constants.py for e.g. of this).